account_payment_terminal/wizards/account_payment_register.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountPaymentRegister(models.TransientModel):
+
+    _name = "account.payment.register"
+    _inherit = ["account.payment.register", "oca.payment.terminal.form.mixin"]
+
+    need_payment_terminal = fields.Boolean(compute="_compute_need_payment_terminal")
+
+    @api.depends("journal_id")
+    def _compute_need_payment_terminal(self):
+        for rec in self:
+            if rec.journal_id.use_payment_terminal:
+                rec.need_payment_terminal = True
+            else:
+                rec.need_payment_terminal = False
```

Comment:
```suggestion
            rec.need_payment_terminal = rec.journal_id.use_payment_terminal
```

?

Reply:
`use_payment_terminal` is a selection field, the ORM will convert `need_payment_terminal` to boolean after assigning a string?

Reply:
Ah, yes, it should convert (or you can write `bool(rec.journal_id.use_payment_terminal)` to be explicit). But from the variable name I honestly thought `use_payment_terminal` was a bool :)